dev/release/download_rc_binaries.py

```diff
@@ -203,7 +203,9 @@ def _download_file(self, dest, asset):
                 "--header",
                 "Accept: application/octet-stream",
                 # Also retry 403s
-                "--retry-all-errors",
+                "--retry",
+                "5",
+                "--fail"
```

Comment:
How about retrying without `--retry-all-errors` when `_download_url()` with `--retry-all-errors` is failed?

```python
try:
  self._download_url(..., "--retry-all-errors", ...)
except ...:
  self._download_url(...)
```

Or we can check `curl` version to confirm whether we can use `--retry-all-errors` or not.

Reply:
Sure. I changed to check the version, though not 100% confident my method is very portable. I welcome suggestions on that.